set.c

```diff
@@ -1913,6 +1913,56 @@ compat_loader(VALUE self, VALUE a)
     return set_i_from_hash(self, rb_ivar_get(a, id_i_hash));
 }
 
+/* C-API functions */
+
+void
+rb_set_foreach(VALUE set, int (*func)(VALUE element, VALUE arg), VALUE arg)
+{
+    set_iter(set, func, arg);
+}
+
+VALUE
+rb_set_new(void)
+{
+    return set_alloc_with_size(rb_cSet, 0);
+}
+
+VALUE
+rb_set_new_capa(unsigned long capa)
```

Comment:
Should this take a `size_t`, to match the type used in `rb_set_size()`?

Reply:
The underlying function takes the size as `st_index_t`. I didn't want to have the public header depend on `st.h` types. I think either `size_t` or `unsigned long` would work. For the common platforms, `st_index_t` is `unsigned long` (I think it's `unsigned long long` on Windows).

Reply:
I think `rb_set_new_capa()` and `rb_set_size()` should use the same type. Avoiding `st_index_t` here seems like a good idea to me as well.

I think `size_t` would be simpler, since we seem to assume it has the same size as a pointer type (and happens to be compatible with `st_index_t`), whereas `unsigned long` can sometimes be smaller.

Reply:
Makes sense to me. When I rebased today, I included that change.